v-next/hardhat/src/internal/cli/main.ts

```diff
@@ -26,6 +33,14 @@ import {
 } from "../helpers/config-loading.js";
 import { getHardhatRuntimeEnvironmentSingleton } from "../hre-singleton.js";
 
+/**
+ * The parsed arguments of a task. This is the result of parsing the raw
+ * arguments passed to a task.
+ */
+interface ParsedTaskArguments extends TaskArguments {
```

Comment:
I'm confused about this again, how is it different from TaskArguments?

Reply:
Maybe the thing that needs to change is TaskArguemtns to have `ParameterValue | ParameterValue[]` instead of `any`.

Reply:
These are the task arguments after they've been parsed by `parseTaskAndArguments`, so they can have better typings for the values. The interface It's only used here. If it's too confusing we could remove it and use `Record<string, ParameterValue | ParameterValue[]>` directly in the function return type instead.

Reply:
But shouldn't we also restrict `TaskArguments` to be `Record<string, ParameterValue | ParameterValue[]>`? that's what I meant.

Reply:
The issue is that we have no guarantees that the arguments have been parsed by the CLI beforehand when the `run` method is called. It could be called programmatically with unknown values. We could restrict it anyway, as for the parameter to be valid, it needs to be a `ParameterValue`.

Reply:
Just to leave the resolution documented: we've decided to use `TaskArguments` over `ParsedTaskArguments` with the improved typings of `Record<string, ParameterValue | ParameterValue[]>`, as any value that does not satisfy this constraint will not be a valid parameter value anyway.